pitch.md

```diff
@@ -1,69 +1,172 @@
-# Project Title (replace me)
+# Compile-Time Programming
 
 ## Overview
-_What's this project about, briefly?_
 
+There are aspects of programming languages that operate during compile time.
+Examples are simple constant expressions, type checking, and macros. Less common
+are dependent types, correctness proofs (relatedly), and LISP-style abstract syntax tree macros.
+
+However, these are features which are extremely difficult to _add_ to an existing language.
+And they are nonetheless so desirable that people will go to extreme lengths to get them
+when they aren't available. (See: C++ template programming, or anything Andrei Alexandrescu writes)
+Furthermore, the nastiness of C++'s templates are not just inherited from C++'s
+general style. In Scala, similar things start to happen when you try to implement
+dependent types.
+
+Ultimately, the real problem is that programmers only have incidental control
+over compile-time behavior. I want to design a language where there is
+direct and encouraged access to compile-time behavior. This is extremely ambitious,
+so there are many limits that will be discussed later.
```

Comment:
Quite ambitious but this would be so cool, as we discussed. Don't you wish there was an API that Java gave you for controlling compile time behavior? Then you could just write a really pretty wrapper 😊